Update mysql.cc for MDEV-15538

chriscalender opened the pull request:
MDEV-15538 (removal of extra <TR> tag in output when -N or --skip-column-names used)

https://jira.mariadb.org/browse/MDEV-15538

Before Fix (note first one uses -N and contains the extra <TR> tag, whereas the latter without -N does not):

```
c:\wpafb\hangar18\10.3\bin>mysql -uroot -pmysql -P3318 -NHBe "select 1"
<TABLE BORDER=1><TR><TR><TD>1</TD></TR></TABLE>
```
```
c:\wpafb\hangar18\10.3\bin>mysql -uroot -pmysql -P3318 -HBe "select 1"
<TABLE BORDER=1><TR><TH>1</TH></TR><TR><TD>1</TD></TR></TABLE>
```

After fix (note that neither now contain the extra <TR> tag):

```
c:\wpafb\hangar18\10.3\bin>mysql -uroot -pmysql -P3318 -NHBe "select 1"
<TABLE BORDER=1><TR><TD>1</TD></TR></TABLE>
```
```
c:\wpafb\hangar18\10.3\bin>mysql -uroot -pmysql -P3318 -NHBe "select 1"
<TABLE BORDER=1><TR><TD>1</TD></TR></TABLE>
```

an3l commented:
I would only suggest to write the test case for this and add it to this PR along with result file. 
Something like this
```
git diff main/mysql.test
diff --git a/mysql-test/main/mysql.test b/mysql-test/main/mysql.test
index c65860eb822..8f08bcda4e1 100644
--- a/mysql-test/main/mysql.test
+++ b/mysql-test/main/mysql.test
@@ -680,3 +680,8 @@ show create table "a1\""b1";
 select * from "a1\""b1";
 drop table "a1\""b1";
 set sql_mode=default;
+
+--echo #
+--echo # MDEV-15538 '-N' Produce html output wrong
+--echo #
+--exec $MYSQL -NHe "select 1 as a"
```
And to change the name of the title in Jira.

chriscalender commented:
Hi Anel,  Thank you for the advice and patience.  I have updated mysql.test and mysql.result accordingly, as well as the Jira title.  (I've spent some time learning 'git' better.  Now to learn more about 'mysql-test'.  Thank you for the example. that was very helpful to get me started, and hopefully I have updated these 2 files correctly.  And for reference, I did run mysql-test-run.pl main.mysql and it passed, and the output matches what is in the main.mysql.result too.)

an3l commented:
Hi Chris,
you can find tutorials we did regarding this :)
Here you find about [submitting a pull request](https://mariadb.org/get-involved/getting-started-for-developers/submitting-pull-request/). 
There you find how to rebase. It would be great to practice on this pull request to rebase your last 3 commits and force pushed it on this branch in order to have only 1 commit :) ? 
There are also tutorial regarding the [writing the good test case for PR](https://mariadb.org/get-involved/getting-started-for-developers/writing-good-test-cases-mariadb-server/)
For your other PRs I suggest you to see [how to get started with code](https://mariadb.org/get-involved/getting-started-for-developers/get-code-build-test/).
After rebasing according to my opinion this PR is ready to be merged.

chriscalender commented:
Hm, well, I tried this but apparently failed somewhere along the line.

I used this to force-push:
git push --force https://github.com/chriscalender/server/ patch-2

The, this page:
https://github.com/MariaDB/server/pull/1167

Said the pull request was "closed" and that 0 files changed and 0 commits.

I will delete the branch and start from fresh.

chriscalender commented:
Here is the new one:
https://github.com/MariaDB/server/pull/1174

Once okay, then I will officially delete this branch.  Leaving open for the notes.

Thank you for those links, btw!

Luuk34 commented:
```
D:\TEMP\server>D:\TEMP\server\bld\client\RelWithDebInfo\mysql.exe -u root --version
D:\TEMP\server\bld\client\RelWithDebInfo\mysql.exe  Ver 15.1 Distrib 10.4.4-MariaDB, for Win32 (AMD64), source revision c0ba036b774b8e35567dccce421890d2199f30ce

D:\TEMP\server>D:\TEMP\server\bld\client\RelWithDebInfo\mysql.exe -u root -NHBe "select 1"
<TABLE BORDER=1>**<TR>**<TR><TD>1</TD></TR></TABLE>
D:\TEMP\server>D:\TEMP\server\bld\client\RelWithDebInfo\mysql.exe -u root -HBe "select 1"
<TABLE BORDER=1><TR><TH>1</TH></TR><TR><TD>1</TD></TR></TABLE>
D:\TEMP\server>
```

The problem is not solved?!

svoj commented:
It is merged up to 10.3 atm, not yet in 10.4.
